### PR TITLE
WIP: PR of changes from #2994 and other changes based on #2897

### PR DESCRIPTION
**Pull request by gkatsev**

### CODE REVIEW

File: src/css/components/menu/_menu-inline.scss [hunk 1]
```diff
@@ -13,6 +13,7 @@
 // Hover state
 .video-js .vjs-menu-button-inline:hover,
 .video-js .vjs-menu-button-inline:focus,
+.video-js .vjs-menu-button-inline[aria-expanded="true"],
```
Comment (dmlap) [inline src/css/components/menu/_menu-inline.scss:16]: My understanding is that attribute queries are still slow in most browsers. Is there a class we could hang this off of?

File: src/js/control-bar/volume-menu-button.js [hunk 2]
```diff
@@ -58,13 +58,42 @@ class VolumeMenuButton extends MenuButton {
...
+    let playerEl = player.el();
+    let handleMouseOver = (event) => {
+      if (event.target !== this.el_ && event.target !== this.volumeBar.el_) {
+        this.el_.setAttribute('aria-expanded', 'false');
+      }
+    };
+    playerEl.onfocusin = handleMouseOver;
+    playerEl.addEventListener('focus',handleMouseOver,true);
+    playerEl.onfocusout = handleMouseOut;
+    playerEl.addEventListener('blur',handleMouseOut,true);
```
Comment (gkatsev) [inline src/js/control-bar/volume-menu-button.js]: This block needs to be cleaned up. I think all the `handleMouseOut` can be removed. Also, it needs to be removed on `dispose`.

File: src/js/menu/menu.js [hunk 3]
```diff
@@ -47,11 +47,15 @@ class Menu extends Component {
     let contentElType = this.options_.contentElType || 'ul';
     this.contentEl_ = Dom.createEl(contentElType, {
       className: 'vjs-menu-content'
+    }, {
+      role: 'listbox'
```
Comment (gkatsev) [inline src/js/menu/menu.js:51]: I don't think we want the changes in this file.

File: src/js/control-bar/volume-menu-button.js [hunk 4]
```diff
@@ -58,13 +58,42 @@ class VolumeMenuButton extends MenuButton {
...
+
+    this.on(player, 'focus', Fn.bind(this, function(event) {
+      console.log('playerfocus', event.target);
```
Comment (gkatsev) [inline src/js/control-bar/volume-menu-button.js]: Oops.

**Comment by gkatsev** [comment]
Heh, @houndci.

File: src/css/components/menu/_menu-inline.scss (hunk 1 above)
Comment (gkatsev) [inline src/css/components/menu/_menu-inline.scss:16]: I don't think that it really makes a difference in most browsers that we care about other than maybe browsers like IE8.
There isn't a class we can use right now and this attribute selector is also technically more semantically correct. We could toggle a class in addition if necessary.

File: src/js/control-bar/volume-menu-button.js [hunk 5]
```diff
@@ -108,11 +137,39 @@ class VolumeMenuButton extends MenuButton {
...
    */
-  handleClick() {
+  handleClick(event) {
     MuteToggle.prototype.handleClick.call(this);
+    this.toggleMutedLabel();
     super.handleClick();
   }
 
+  toggleMutedLabel() {
+    if (this.player_.muted()) {
```
Comment (gkatsev) [inline src/js/control-bar/volume-menu-button.js:147]: the specific labels here probably need to be tweaked. Maybe just saying `Muted` and `Unmuted` is enough. (Localized, of course).

**Pull request closed without merging**